src/netconf_acm.c

```diff
@@ -0,0 +1,1833 @@
+                        rule->target_type = SR_NACM_TARGET_DATA;
+                    }
+                }
+            } else if (!strcmp(node->schema->name, "access-operations")) {
+                str = lyd_get_value(node);
+                rule->operations = 0;
+                if (!strcmp(str, "*")) {
+                    rule->operations = SR_NACM_OP_ALL;
+                } else {
+                    if (strstr(str, "create")) {
+                        rule->operations |= SR_NACM_OP_CREATE;
+                    }
+                    if (strstr(str, "read")) {
+                        rule->operations |= SR_NACM_OP_READ;
+                    }
+                    if (strstr(str, "update")) {
+                        rule->operations |= SR_NACM_OP_UPDATE;
+                    }
+                    if (strstr(str, "delete")) {
+                        rule->operations |= SR_NACM_OP_DELETE;
+                    }
+                    if (strstr(str, "exec")) {
+                        rule->operations |= SR_NACM_OP_EXEC;
+                    }
+                }
+            } else if (!strcmp(node->schema->name, "action")) {
+                if (!strcmp(lyd_get_value(node), "permit")) {
+                    rule->action_deny = 0;
+                } else {
+                    rule->action_deny = 1;
+                }
+            } else if (!strcmp(node->schema->name, "comment")) {
+                if (op == SR_OP_DELETED) {
+                    free(rule->comment);
+                    rule->comment = NULL;
+                } else {
+                    assert((op == SR_OP_MODIFIED) || (op == SR_OP_CREATED));
+                    free(rule->comment);
+                    rule->comment = strdup(lyd_get_value(node));
+                }
+            }
+        }
+    }
+
+    /* NACM UNLOCK */
+    pthread_mutex_unlock(&nacm.lock);
+
+    sr_free_change_iter(iter);
+    if (rc != SR_ERR_NOT_FOUND) {
+        sr_session_set_error_message(session, "Getting next change failed (%s).", sr_strerror(rc));
+        return rc;
+    }
+
+    return SR_ERR_OK;
+}
+
+API void
+sr_nacm_init(void)
+{
+    pthread_mutex_init(&nacm.lock, NULL);
+}
+
+#define SR_CONFIG_SUBSCR(session, mod_name, xpath, cb) \
+    rc = sr_module_change_subscribe(session, mod_name, xpath, cb, NULL, 0, \
+            SR_SUBSCR_DONE_ONLY | SR_SUBSCR_ENABLED, sub); \
+    if (rc != SR_ERR_OK) { \
+        sr_errinfo_new(&error_info, rc, "Subscribing for \"%s\" data changes failed.", mod_name); \
+        return error_info; \
+    }
+
+#define SR_OPER_SUBSCR(session, mod_name, xpath, cb) \
+    rc = sr_oper_get_subscribe(session, mod_name, xpath, cb, NULL, 0, sub); \
+    if (rc != SR_ERR_OK) { \
+        sr_errinfo_new(&error_info, rc, "Subscribing for providing \"%s\" state data failed.", mod_name); \
+        return error_info; \
+    }
+
+API sr_error_info_t *
+sr_nacm_init_callbacks(sr_session_ctx_t *session, sr_subscription_ctx_t **sub)
```

Comment:
Is there a reason why this is not part of `sr_nacm_init()`? Also, it would probably be useful to allow using `NO_THREAD` flag for the subscription.

Reply:
No, there is no particular reason. I have merged the functions. I have also added custo subscription options